File: Exercism/javascript/resistor-color-duo/resistor-color-duo.js
//
// This is only a SKELETON file for the 'Resistor Color Duo' exercise. It's been provided as a
// convenience to get you started writing code faster.
//


let COLORS = ["black","brown","red","orange","yellow","green","blue","violet","grey","white"];

// export const value = (arr) => {
//   let value1 = COLORS.indexOf(arr[0]).toString();
//   let value2 = COLORS.indexOf(arr[1]).toString();
//    return Number(value1 + value2);
// };

// export const value = ([str1,str2]) => {
//   let value1 = COLORS.indexOf(str1).toString();
//   let value2 = COLORS.indexOf(str2).toString();
//    return Number(value1 + value2);
// };

export const value = array => Number(array.map(item => COLORS.indexOf(item)).join(''));

File: Exercism/ruby/resistor-color-duo/resistor_color_duo.rb
class ResistorColorDuo
    def self.value(colors)
        
        colors = colors.map(&:capitalize)
              
        color_value = { "Black" => 0,
                        "Brown"=> 1,
                        "Red"=> 2,
                        "Orange"=> 3,
                        "Yellow"=> 4,
                        "Green"=> 5,
                        "Blue"=> 6,
                        "Violet"=> 7,
                        "Grey"=> 8,
                        "White"=> 9
                    }

                
        colors.map { |color| color_value[color].to_s}
              .join
              .to_i
        
    end
end
